.script/SolutionValidations/validDomainsVerticals.ts

```diff
@@ -0,0 +1,102 @@
+    if (filePath.endsWith("mainTemplate.json")) {
+        // read the content of the file
+        let jsonFile = JSON.parse(fs.readFileSync(filePath, "utf8"));
+
+        // check if the file has a "resources" field
+        if (!jsonFile.hasOwnProperty("resources")) {
+            console.warn(`No "resources" field found in the file. Skipping file path: ${filePath}`);
+            return ExitCode.SUCCESS;
+        }
+
+        // get the resources from the file
+        let resources = jsonFile.resources;
+
+        // filter resources that have type "Microsoft.OperationalInsights/workspaces/providers/metadata"
+        const filteredResource = resources.filter(function (resource: { type: string; }) {
+            return resource.type === "Microsoft.OperationalInsights/workspaces/providers/metadata";
+        });
+        if (filteredResource.length > 0) {
+            filteredResource.forEach((element: { hasOwnProperty: (arg0: string) => boolean; properties: { hasOwnProperty: (arg0: string) => boolean; categories: any; }; }) => {
+                // check if the resource has a "properties" field
+                if (element.hasOwnProperty("properties") === true) {
+                    // check if the "properties" field has a "categories" field
+                    if (element.properties.hasOwnProperty("categories") === true) {
+                        const categories = element.properties.categories;
+
+                        let invalidDomains = [];
+                        let invalidVerticals = [];
+
+                        // check if the categories have a "domains" field
+                        if (categories.hasOwnProperty("domains")) {
+                            let domains = categories.domains;
+                            if (domains.length === 0) {
+                                throw new MainTemplateDomainVerticalValidationError("The solution must include at least one valid domain. Please provide a domain in the 'domains' field of the 'categories' object.");
+                            }
+                            for (const domain of domains) {
+                                // check if the domain is valid
+                                if (!validDomains.includes(domain)) {
+                                    invalidDomains.push(domain);
+                                }
+                            }
+                        }
+                        else {
+                            throw new MainTemplateDomainVerticalValidationError("The solution must include at least one valid domain. Please provide a domain in the 'domains' field of the 'categories' object.");
+                        }
+
+                        // check if the categories have a "verticals" field
+                        if (categories.hasOwnProperty("verticals")) {
+                            let verticals = categories.verticals;
+                            for (const vertical of verticals) {
```

Comment:
Can we add empty verticals array condition? means it has verticals property but doesn't have any array value in that.

Reply:
I think it is not necessary here, because if they provide empty vertical array then it will fail for arm-ttk validation